### ISSUE TITLE
Build failure on GHC 8.0.1 and Cabal 1.24.0.0

### ISSUE THREAD

**Issue by hellerve** [ryantm/hdbc-mysql#21]
While compiling HDBC-mysql (on OS X 10.11.6, mysql 14.14, GHC and Cabal specified above) I encountered this error:

```
/var/folders/h7/xm3lt1wn0hg50s4bfkqy3ht00000gr/T/cabal-tmp-54890/HDBC-mysql-0.6.6.2/dist/setup/setup.hs:35:7:
error:
• Couldn't match type ‘[Char]’ with ‘(FilePath, [FilePath])’
Expected type: IO (Maybe (FilePath, [FilePath]))
Actual type: IO (Maybe FilePath)
• In a stmt of a 'do' block:
return (mysql_config `mplus` mysql_config5)
In the expression:
do { mysql_config <- findProgramLocation verbosity "mysql_config";
mysql_config5 <- findProgramLocation verbosity "mysql_config5";
return (mysql_config `mplus` mysql_config5) }
In the ‘programFindLocation’ field of a record
)
```

I am unsure whether this is enough information as I am unsure where the typechecker chokes.
If you need any more info, I am more than happy to provide it if I can.

Cheers

**Comment by ryantm** [comment]
Hello.

Thanks for your report!

Please don't use 0.6.6.2 on 8.0.1. The 0.6 series is not compatible with 8.0.1, and I've adjusted the bounds of base in 0.6.6.4 to reflect that. I just released 0.7.0.0, which should work with 8.0.1. Please try it and let me know if you have any issues!

Ryan

**Comment by hellerve** [comment]
Hey Ryan, 

Thanks for the quick reply. I'm going to test it out right away and tell you if I have any issues. Best case is you won't hear from me anymore! 

Cheers
